scripts/parse_anglerfish_results.py

```diff
@@ -93,127 +104,73 @@ def parse_data(df_raw: pd.DataFrame):
         else None,
         axis=1,
     )
+    # Get barcode number from ID
+    df["ont_barcode_id"] = df["ont_barcode"].apply(
+        lambda x: int(str(x)[-2:]) if pd.notna(x) else None
+    )
 
     return df
 
 
-def ont_barcode_well2name(barcode_well: str) -> str:
-    # Add colon if not present
-    if ":" not in barcode_well:
-        barcode_well = f"{barcode_well[0]}:{barcode_well[1:]}"
-
-    # Get the number corresponding to the well (column-wise)
-    barcode_num_str = str(formula.well_name2num_96plate[barcode_well])
-
-    # Pad barcode number with leading zero if necessary
-    if len(barcode_num_str) < 2:
-        barcode_num_str = f"0{barcode_num_str}"
-    barcode_name = f"barcode{barcode_num_str}"
-
-    return barcode_name
-
-
-def fill_udfs(currentStep: Process, df: pd.DataFrame):
-    # Get Illumina pools
-    illumina_pools = [
-        input_art
-        for input_art in currentStep.all_inputs()
-        if input_art.type == "Analyte"
-    ]
-
-    for illumina_pool in illumina_pools:
+def fill_udfs(process: Process, df: pd.DataFrame):
+    errors = False
+
+    # Get Illumina samples
+    measurements = []
+    ops = process.all_outputs()
+    for op in ops:
+        if op.name in list(df.sample_name) and len(op.samples) == 1:
+            measurements.append(op)
+    measurements.sort(key=lambda x: x.name)
+
+    assert len(measurements) == len(
+        df["sample_name"].isin([m.name for m in measurements])
+    ), "Number of samples demultiplexed in LIMS does not correspond to \
+    number of sample rows in Anglerfish results."
+
+    # Relate UDF names to dataframe column names
+    udf2col = {
+        "# Reads": "num_reads",
+        "Avg. Read Length": "mean_read_len",
+        "Std. Read Length": "std_read_len",
+        "Representation Within Run (%)": "repr_total_pc",
+        "Representation Within Barcode (%)": "repr_within_barcode_pc",
+        "ONT Barcode ID": "ont_barcode_id",
+    }
+
+    for measurement in measurements:
+        sample_name = measurement.name
+        sample_row = df[df["sample_name"] == sample_name]
+
+        # Assign UDFs
+        for udf, col in udf2col.items():
+            try:
+                if pd.notna(sample_row[col].values[0]):
+                    value = float(sample_row[col].values[0])
+                    measurement.udf[udf] = value
+            except:
+                errors = True
+                logging.error(
```

Comment:
Perhaps we want to log what the python error is here? The bare except could hide a lot of nasty things that would go unnoticed otherwise. But I guess it's not something that the user would like to see? Maybe catch the specific exceptions that you are expecting first and then only if it's not expected the actual error would be logged? Alternatively fail completely if it's an unexpected exception

Reply:
These blocks are quite limited in their executive scope. One only assigns measurement UDFs and the subsequent block attempts to write the updated measurement to LIMS.

I've added traceback to the error messages in 36ee3ef2023e8b0da87e541ee86f9ae1c913c44c, though.

Reply:
You don't know which exceptions you're expecting? As a reviewer but even more as a future developer of this part of the code, I have no idea why the bare except is there as it is now.

Reply:
How about this?

6f6f20da6f337a289124e29429348cddcd565a54